src/App.tsx

```diff
@@ -1,39 +1,66 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
 
+import { Clock } from './Clock';
+
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+type State = {
+  clockName: string,
+  hasClock: boolean
+};
+
+export class App extends Component<{}, State> {
+  state = {
+    clockName: 'clock-0',
+    hasClock: true,
+  };
+
+  timerId: number | null = null;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      const clockName = getRandomName();
+
+      this.setState({ clockName });
+    }, 3300);
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+    document.addEventListener('click', this.handleClick);
+    document.addEventListener('contextmenu', this.handleRightClick);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  componentWillUnmount() {
+    if (this.timerId) {
+      window.clearInterval(this.timerId);
+    }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    document.removeEventListener('click', this.handleClick);
+    document.removeEventListener('contextmenu', this.handleRightClick);
+  }
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  handleClick = () => {
```

Comment:
you want your naming consistent to have it maximally clear
```suggestion
  handleLeftClick = () => {
```